kotest-assertions-arrow-fx-coroutines/src/commonMain/kotlin/io/kotest/assertions/arrow/fx/coroutines/ResourceExtension.kt

```diff
@@ -0,0 +1,170 @@
+package io.kotest.assertions.arrow.fx.coroutines
+
+import arrow.fx.coroutines.ExitCase
+import arrow.fx.coroutines.Resource
+import io.kotest.core.extensions.LazyMaterialized
+import io.kotest.core.extensions.LazyMountableExtension
+import io.kotest.core.listeners.AfterSpecListener
+import io.kotest.core.listeners.TestListener
+import io.kotest.core.spec.Spec
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestResult
+import io.kotest.core.test.TestType
+import io.kotest.core.test.isRootTest
+import io.kotest.mpp.atomics.AtomicReference
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.DelicateCoroutinesApi
+
+public enum class LifecycleMode {
+  Spec, EveryTest, Leaf, Root
+}
+
+public fun <A> Resource<A>.extension(lifecycleMode: LifecycleMode = LifecycleMode.Spec): ResourceExtension<A> =
+  ResourceExtension(this, lifecycleMode)
+
+/**
+ * Ensures that all finalizers of [resource] are invoked for the specified [LifecycleMode].
+ * Installing this [LazyMountableExtension] returns a [LazyMaterialized]
+ * from which you can extract initialized [A] using [LazyMaterialized.get].
+ *
+ * ```kotlin
+ * class Connection(val name: String) {
+ *   fun open() = println("Open $name")
+ *   fun compute() = "Result of $name"
+ *   fun release(ex: ExitCase) = println("Closing $name with $ex")
+ * }
+ *
+ * fun connection(name: String): Resource<Connection> =
+ *   Resource(
+ *     { Resource(name) },
+ *     { conn, ex -> conn.release(ex) }
+ *   )
+ *
+ * class ResourceSpec : StringSpec({
+ *   val conn: TestResource<Connection> =
```

Comment:
Is `TestResource` still a thing? Or is this a `ProjectTestResource` now?

Reply:
TestResource is no longer a thing! Will update the docs. Where is that page coming from? Main kotest repo?